pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/SnapshotSegmentAbortedTxnProcessorImpl.java

```diff
@@ -265,7 +266,7 @@ public CompletableFuture<PositionImpl> recoverFromSnapshot() {
                     PositionImpl finalStartReadCursorPosition = startReadCursorPosition;
                     TransactionBufferSnapshotIndexes finalPersistentSnapshotIndexes = persistentSnapshotIndexes;
                     if (persistentSnapshotIndexes == null) {
```

Comment:
After turning on the feature `transactionBufferSegmentedSnapshotEnabled,` if users create a new topic, the method `recoverOldSnapshot` will trigger the creation of the topic `__transaction_buffer_snapshot`, but it is not needed, right?

And can we add a test to guarantee this issue will not occur?